tests/IntegrationTests/WebAppUiTests/UiTestHelpers.cs

```diff
@@ -1,70 +1,275 @@
-        /// <param name="page">The page object whose context is recording a trace</param>
-        /// <returns>Nothing just creates the file</returns>
-        public static async Task EndAndWritePlaywrightTrace(IPage page) 
-        {
-            await page.Context.Tracing.StopAsync(new()
-            {
-                Path = "PlaywrightTrace.zip"
-            });
-        }
-    }
-}
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Management;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Microsoft.Identity.Web.Test.Common;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.Playwright;
+using Microsoft.VisualStudio.TestPlatform.Utilities;
+using Xunit.Abstractions;
+
+namespace WebAppUiTests
+{
+    public static class UiTestHelpers
+    {
+        public static async Task FirstLogin_MicrosoftIdentityFlow_ValidEmailPassword(IPage page, string email, string password, ITestOutputHelper? output=null ,bool staySignedIn=false)
+        {
+            string staySignedInText = staySignedIn ? "Yes" : "No";
+
+            WriteLine(output, $"Logging in ... Entering and submitting user name: {email}");
+            ILocator emailInputLocator = page.GetByPlaceholder(TestConstants.EmailText);
+            await FillEntryBox(emailInputLocator, email);
+            await EnterPassword_MicrosoftIdentityFlow_ValidPassword(page, password, staySignedInText);
+        }
+
+        public static async Task SuccessiveLogin_MicrosoftIdentityFlow_ValidEmailPassword(IPage page, string email, string password, ITestOutputHelper? output = null, bool staySignedIn = false)
+        {
+            string staySignedInText = staySignedIn ? "Yes" : "No";
+
+            WriteLine(output, $"Logging in again in this browsing session... selecting user via email: {email}");
+            await SelectKnownAccountByEmail_MicrosoftIdentityFlow(page, email);
+            await EnterPassword_MicrosoftIdentityFlow_ValidPassword(page, password, staySignedInText);
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="page"></param>
+        /// <param name="email"></param>
+        /// <param name="signOutPageUrl"></param>
+        /// <returns></returns>
+        public static async Task PerformSignOut_MicrosoftIdentityFlow(IPage page, string email, string signOutPageUrl, ITestOutputHelper? output = null)
+        {
+            WriteLine(output, "Signing out ...");
+            await SelectKnownAccountByEmail_MicrosoftIdentityFlow(page, email);
+            await page.WaitForURLAsync(signOutPageUrl);
+            WriteLine(output, "Sign out page successfully reached");
+        }
+
+        /// <summary>
+        /// In the Microsoft Identity flow, the user is at certain stages presented with a list of accounts known in 

```

Comment:
>Microsoft Identity flow [](http://example.com/codeflow?start=19&length=23)

this doesn't exist...at least there is no spec for it...IT's just the web app sign-in.

Reply:
@jennyf19 I was trying to refer to the fact that we are signing in using ESTS login flow, my thought is there are other login providers that exist, and this makes it clear which we are using. Is there a better name for it since ESTS isn't an external term?